dauthz/core.py

```diff
@@ -1,13 +1,46 @@
+# from casbin_adapter.adapter import Adapter as CasbinAdapter
+import logging
+
 from casbin import Enforcer
+from django.conf import settings
+
+from .utils import import_class
 
 
 class ProxyEnforcer(Enforcer):
-    def __init__(self):
-        # todo read configuration, initialize Enforcer...
-        model = None
-        adapter = None 
-        super.__init__(model, adapter)
-        pass
+    name = None
+    logger = None
+    enable_log = None
+
+    def __init__(self, enforcer_name: str, enforcer_conf: dict):
+        self.name = enforcer_name
+        self.logger = logging.getLogger(self.name)
+
+        model_type = enforcer_conf["MODEL"]["CONFIG_TYPE"]
+        if model_type == "file":
+            model = enforcer_conf["MODEL"]["CONFIG_FILE_PATH"]
+        else:
+            model = enforcer_conf["MODEL"]["CONFIG_TEXT"]
+        model = str(model)
+        self.enable_log = enforcer_conf["LOG"]["ENABLED"]
+
+        adapter_loc = enforcer_conf["ADAPTER"]["NAME"]
+        adapter_class = import_class(adapter_loc)
+        adapter = adapter_class()
+        super().__init__(model, adapter)
+
+        watcher = enforcer_conf.get("WATCHER")
+        if watcher:
+            self.set_watcher(watcher)
+        role_manager = enforcer_conf.get("ROLE_MANAGER")
+        if role_manager:
+            self.set_role_manager(role_manager)
+        self.logger.debug("Casbin enforcer initialised")
 
 
-enforcer = ProxyEnforcer()
```

Comment:
There should be a default enforcer here.

Reply:
![image](https://user-images.githubusercontent.com/75596353/184543467-a6ea8c7a-7459-438a-a5c0-68b624cd6ca1.png)
![image](https://user-images.githubusercontent.com/75596353/184543518-5acfe8b3-b29b-4438-a673-5898d875dba4.png)
The default enforcer is specified by passing the DEFAULT field, but the user can also modify the enforcer as needed